homeassistant/components/sensor/__init__.py

```diff
@@ -178,10 +180,12 @@ def state_attributes(self) -> dict[str, Any] | None:
                     self.state_class,
                     report_issue,
                 )
+            attrs[ATTR_LAST_RESET] = last_reset.isoformat()
 
-            return {ATTR_LAST_RESET: last_reset.isoformat()}
+        if native_value := self.native_value:
+            attrs[ATTR_NATIVE_VALUE] = native_value
```

Comment:
I'm not sure we should add the native value to the state machine. We don't do this currently for any of the existing cases where there are conversions in the base entity integration, eg temperature Celsius to Fahrenheit , `is_on` to `STATE_ON` or `STATE_OFF`. Currently we let each integration handle the conversion back to native value. We should probably look at this on an application level and decide how this should be done generally.